cloudbridge/cloud/providers/gce/services.py

```diff
@@ -267,16 +270,122 @@ def current(self):
         return self.get(self.provider.region_name)
 
 
+class GCEImageService(BaseImageService):
+
+    def __init__(self, provider):
+        super(GCEImageService, self).__init__(provider)
+        self._public_images = None
+
+    _PUBLIC_IMAGE_PROJECTS = ['centos-cloud', 'coreos-cloud', 'debian-cloud',
+                             'opensuse-cloud', 'ubuntu-os-cloud']
+
+    def _retrieve_public_images(self):
+        if self._public_images is not None:
+            return
+        self._public_images = []
+        for project in GCEImageService._PUBLIC_IMAGE_PROJECTS:
+            try:
+                response = self.provider.gce_compute \
+                                        .images() \
+                                        .list(project=project) \
+                                        .execute()
+            except googleapiclient.errors.HttpError as http_error:
+                cb.log.warning("googleapiclient.errors.HttpError: {0}".format(
+                    http_error))
+            if 'items' in response:
+                self._public_images.extend(
+                    [GCEMachineImage(self.provider, image) for image
+                     in response['items']])
+
+    def get(self, image_name):
```

Comment:
I see, so the GCE APIs use 'name' as the external primary key, and 'id' as some internal pk value which is not intended for public consumption? In that case, it may make sense to make the GCEMachineImage.id function return self._gce_image['name'] instead of self._gce_image['id']. This is because in CloudBridge, the id refers to the primary key through which an object can be retrieved.  Subsequently, the get method's signature can also use the image_id?